tools/hlstool/hlstool.cpp

```diff
@@ -109,6 +109,22 @@ static cl::opt<HLSFlow>
                        clEnumValN(HLSFlowDynamicHW, "dynamic-hw",
                                   "Dynamically scheduled (HW path)")));
 
+enum DynamicParallelismKind {
+  DynamicParallelismNone,
+  DynamicParallelismLocking,
+  DynamicParallelismPipelining
+};
+
+static cl::opt<DynamicParallelismKind> dynParallelism(
+    "dynamic-parallelism", cl::desc("Specify the DHLS task parallelism kind"),
+    cl::values(clEnumValN(DynamicParallelismNone, "none",
+                          "Add no protetion mechanisms"),
```

Comment:
protetion -> protection.

Description could probably be elaborated slightly - what is implied by having no protection mechanisms?